sdk/core/src/services/github/service.rs

```diff
@@ -0,0 +1,102 @@
+
+use crate::services::github::error::Error;
+use crate::services::github::client::{
+    Client as GitHubClient,
+    Repo,
+};
+
+/// Definition of the GitHubProvider
+#[derive(Debug)]
+pub struct GitHubService {
+    org: String,
+    client: GitHubClient,
+}
+
+/// Impl for GitHubSbomProvider
+///
+impl GitHubService {
+    /// new method sets the organization for the struct
+    ///
+    pub fn new(org: String, pat: String) -> Self {
+        let client = GitHubClient::new(pat);
+        GitHubService {
+            org,
+            client,
+        }
+    }
+
+    /// Use the GitHub Client to get all of the repositories
+    /// from the GitHub Organization
+    pub(crate) async fn get_repos(&self) -> Result<Vec<Repo>, Error> {
+
+        let mut pages = match self.client.get_pages(&self.org).await {
+            Ok(pages) => pages,
+            Err(err) => panic!("Unable to get pages of repos from GitHub: {:#?}", err)
+        };
+
+        let mut repo_vec: Vec<Repo> = Vec::new();
+        for (page, per_page) in pages.iter_mut().enumerate() {
+
+            let mut gh_org_rsp = self.client.get_page_of_repos(&self.org, page + 1, per_page).await?;
+
+            for (repo_num, mut repo) in gh_org_rsp.iter_mut().enumerate() {
+
+                print!("Repo number: {}, ", repo_num);
```

Comment:
Should be println!. It's not flushing and that's weird.  Probably should just remove it.